tests/app/main/test_services.py

```diff
@@ -1989,28 +1989,49 @@ def test_remove_subsection(self, s3):
             'digital-specialists/1/remove/individual-specialist-roles/agile-coach'
         )
 
-        assert res.status_code == 302
-        assert(
-            '/suppliers/frameworks/digital-outcomes-and-specialists/submissions/digital-specialists/1?' in res.location
+        assert res.status_code == 200
+        doc = html.fromstring(res.get_data(as_text=True))
+
+        page_title = doc.xpath('//h1')[0].text_content()
+        title_caption = doc.cssselect('span.govuk-caption-xl')[0].text_content()
+        assert title_caption == "Digital specialists - Individual specialist roles"
+        assert page_title == "Are you sure you want to remove agile coach?"
+
+        assert len(
+            doc.cssselect(
+                'form[action="/suppliers/frameworks/digital-outcomes-and-specialists/submissions/' +
+                'digital-specialists/1/remove/individual-specialist-roles/agile-coach"]'
+            )
+        ) == 1
+
+        submit_button = doc.cssselect('button[name="remove_confirmed"]')
+        cancel_link = doc.xpath("//a[normalize-space()='Cancel']")
+
+        assert len(submit_button) == 1
+        assert len(cancel_link) == 1
```

Comment:
This test seems to have a lot of asserts testing the contents of the page. What sorts of bugs would cause these to fail?

Reply:
Got rid of this test and replaced it with a simpler one to make sure the confirmation pages are loaded when submitting the relevant form.